OrbitQt/ConnectToStadiaWidget.h

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_QT_CONNECT_TO_STADIA_WIDGET_H_
+#define ORBIT_QT_CONNECT_TO_STADIA_WIDGET_H_
+
+#include <QSettings>
+#include <QStateMachine>
+#include <QVector>
+#include <QWidget>
+#include <memory>
+
+#include "ConnectionConfiguration.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitGgp/Client.h"
+#include "OrbitGgp/Instance.h"
+#include "OrbitGgp/InstanceItemModel.h"
+#include "OrbitGgp/SshInfo.h"
+#include "ui_ConnectToStadiaWidget.h"
+
+namespace orbit_qt {
+
+class ConnectToStadiaWidget : public QWidget {
+  Q_OBJECT
+  Q_PROPERTY(bool active READ IsActive WRITE SetActive)
+
+ public:
+  explicit ConnectToStadiaWidget(QWidget* parent = nullptr);
+  // This needs to be called before this class can be used. (It is not part of the constructor,
+  // because otherwise this class could not be instantiated from a .ui file)
+  void SetStadiaConnection(StadiaConnection* stadia_connection);
```

Comment:
In that case we should go the Qt-way which means the widget should be able to work without crashing when no StadiaConnection is set. Just like a QTableView works with no model set.

Maybe that's already the case. I'm not sure.

Reply:
This now all changed a little bit, but the general Idea of the widget not crashing when these setters are not called is done.  Its a pretty useless widget then, but it does not crash :)